packages/patterns/src/components/allow-overrides-modal.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	Button,
+	TextControl,
+	Modal,
+} from '@wordpress/components';
+import { __, sprintf } from '@wordpress/i18n';
+import { useState, useId } from '@wordpress/element';
+import { speak } from '@wordpress/a11y';
+
+export default function AllowOverridesModal( {
+	placeholder,
+	onClose,
+	onSave,
+} ) {
+	const [ editedBlockName, setEditedBlockName ] = useState( '' );
+	const descriptionId = useId();
+
+	const isNameValid = !! editedBlockName.trim();
+
+	const handleSubmit = () => {
+		const message = sprintf(
+			/* translators: %s: new name/label for the block */
+			__( 'Block name changed to: "%s".' ),
+			editedBlockName
+		);
+
+		// Must be assertive to immediately announce change.
+		speak( message, 'assertive' );
+		onSave( editedBlockName );
+
+		// Immediate close avoids ability to hit save multiple times.
+		onClose();
```

Comment:
There seems to be a focus loss here because of the way the button changes to a toggle 🤔

Reply:
Should be fixed now.